src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -1409,12 +1409,12 @@ because xbuild doesn't support framework reference assemblies.
 		ToolExe="$(Aapt2ToolExe)"
 	/>
 	<Touch
-			Files="$(_AndroidLibraryFlatArchivesDirectory)%(_LibraryResourceHashDirectories.Hash).stamp"
+			Files="@(_LibraryResourceHashDirectories->'$(_AndroidLibraryFlatArchivesDirectory)%(Hash).stamp')"
 			AlwaysCreate="True"
 	/>
 	<ItemGroup>
-		<FileWrites Include="$(_AndroidLibraryFlatArchivesDirectory)%(_LibraryResourceHashDirectories.Hash).flata" />
-		<FileWrites Include="$(_AndroidLibraryFlatArchivesDirectory)%(_LibraryResourceHashDirectories.Hash).stamp" />
+		<FileWrites Include="@(_LibraryResourceHashDirectories->'$(_AndroidLibraryFlatArchivesDirectory)%(Hash).flata')" />
+		<FileWrites Include="@(_LibraryResourceHashDirectories->'$(_AndroidLibraryFlatArchivesDirectory)%(Hash).stamp')" />
```

Comment:
So we have a lot of repeated transforms here such as:
```
@(_LibraryResourceHashDirectories->'$(_AndroidLibraryFlatArchivesDirectory)%(Hash).stamp')
```
Could we clean this up a bit by storing them in an intermediate property? I think MSBuild evaluates these every time, so it's doing the work over and over.